application-licensing-licensor/application-licensing-licensor-api/src/main/java/com/xwiki/licensing/internal/upgrades/notifications/ExtensionAutoUpgradedEventConverter.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.licensing.internal.upgrades.notifications;
+
+import static java.util.Collections.singletonList;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.eventstream.Event;
+import org.xwiki.eventstream.RecordableEvent;
+import org.xwiki.eventstream.RecordableEventConverter;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+
+/**
+ * Add additional information to ExtensionAutoUpgradedEvent.
+ *
+ * @version $Id$
+ * @since 1.17
+ */
+@Singleton
+@Named(ExtensionAutoUpgradedEventConverter.NAME)
+@Component
+public class ExtensionAutoUpgradedEventConverter implements RecordableEventConverter
+{
+    /**
+     * The name of this component.
+     */
+    public static final String NAME = "ExtensionAutoUpgradedEventConverter";
+
+    @Inject
+    private RecordableEventConverter defaultConverter;
+
+    @Inject
+    @Named("explicit")
+    private DocumentReferenceResolver<String> explicitDocumentReferenceResolver;
+
+    @Override
+    public Event convert(RecordableEvent recordableEvent, String source, Object data) throws Exception
+    {
+        String upgradeMessage = (String) data;
+        Event convertedEvent = this.defaultConverter.convert(recordableEvent, source, data);
+
+        DocumentReference docRef =
+            this.explicitDocumentReferenceResolver.resolve("Licenses.Code.LicensingConfig", convertedEvent.getUser());
+        convertedEvent.setDocument(docRef);
```

Comment:
Would be good to add a comment to explain what this is about and why you choose to use ``LicensingConfig`` and not something else.